layers/ports.go

```diff
@@ -61,20 +61,20 @@ func (a UDPPort) String() string {
 //
 // Returns gopacket.LayerTypePayload for unknown/unsupported port numbers.
 func (a UDPPort) LayerType() gopacket.LayerType {
-	switch a {
-	case 53:
-		return LayerTypeDNS
-	case 123:
-		return LayerTypeNTP
-	case 4789:
-		return LayerTypeVXLAN
-	case 67, 68:
-		return LayerTypeDHCPv4
-	case 6343:
-		return LayerTypeSFlow
-	default:
-		return gopacket.LayerTypePayload
+	lt := udpPortLayerType[uint16(a)]
+	if lt != 0 {
+		return lt
 	}
+	return gopacket.LayerTypePayload
+}
+
+var udpPortLayerType [65536]gopacket.LayerType
```

Comment:
Note, there's already some complaints about init taking up too much memory (https://github.com/google/gopacket/issues/165).  I'm considering the performance hit of making everything maps instead of arrays.  I'd say this is a fine way to do things for now, though, just something to keep in mind.